### DISCUSSION TITLE
Close transient when clicking outside it

### DISCUSSION THREAD

**Discussion by josephmturner** [magit/transient#322] [Ideas]
When a transient menu is open, clicking outside it gives the error: `Unbound suffix: ‘<mouse-1>’ (Use ‘C-g’ to abort, ‘?’ for help) [mouse-set-point]`.  Would it make sense to run `transient-quit-all` instead?  I think this might especially help new users.

Thanks!

Joseph

**Comment by tarsius** [comment]
In some cases yes, but it becomes more complicated when we take into account that some menus allow invoking arbitrary non-suffix commands. Clicks inside the menu also have to be considered, both when menu navigation is enabled and when it is disabled. We also have to improve `handle-switch-frame` events, which are triggered by mouse movement (depending on window manager and/or Emacs configuration).

I think all this, and probably more, should be considered at the same time, and I don't have the time for that now.

(Same for the suggestion/request for help you recently sent by mail. If you re-post that as a discussion in the Magit repository, then Kyle might choose to help you. And if not, that would at least reduce the risk of me forgetting about it.)

**Reply by josephmturner** [reply]
Thanks for your reply!  I understand that there are many considerations relating to mouse events and transient.

I've made a [new Github discussion](https://github.com/magit/magit/discussions/5249) about the Magit issue.
